### ISSUE TITLE
Redesign SchedulerCommonTest to use less hardcoded timesteps

### ISSUE THREAD

**Issue by dg845** [huggingface/diffusers#5468]
**What API design would you like to have changed or added to the library? Why?**

I'm not sure if this counts as an API design issue but I think it would be more convenient if the tests and utility functions in `SchedulerCommonTest` had less dependency on hardcoded timesteps. For example, some tests that use hardcoded timesteps are as follows:

- [`test_from_save_pretrained`](https://github.com/huggingface/diffusers/blob/5366db5df1fb16bd92491ec2b624d619b21b62a5/tests/schedulers/test_schedulers.py#L411): uses hardcoded timestep `1`
- [`test_step_shape`](https://github.com/huggingface/diffusers/blob/5366db5df1fb16bd92491ec2b624d619b21b62a5/tests/schedulers/test_schedulers.py#L497): uses hardcoded timesteps `0` and `1`
- [`test_scheduler_outputs_equivalence`](https://github.com/huggingface/diffusers/blob/5366db5df1fb16bd92491ec2b624d619b21b62a5/tests/schedulers/test_schedulers.py#L533): uses hardcoded timestep `0`

It's hard to reuse these tests when a new scheduler's `set_timesteps` method typically does not produce `timestep` schedules with 0 or 1 in them. (An example is [`LCMScheduler`](https://github.com/huggingface/diffusers/blob/e5168588864d72a4dca37e90318c6b11da0eaaf1/examples/community/latent_consistency_txt2img.py#L394) in the latent consistency models community pipeline.) For these schedulers the above tests either have to be overridden or changed directly to add logic specific to the scheduler, which is a hassle.

I think it would be better if the tests were able to get a valid `timestep` on the fly. For example, `test_scheduler_outputs_equivalence` hardcodes the `timestep` as `0`:

https://github.com/huggingface/diffusers/blob/5366db5df1fb16bd92491ec2b624d619b21b62a5/tests/schedulers/test_schedulers.py#L563-L565

but it later goes on to call `scheduler.set_timesteps` before it calls `scheduler.step`:

https://github.com/huggingface/diffusers/blob/5366db5df1fb16bd92491ec2b624d619b21b62a5/tests/schedulers/test_schedulers.py#L587-L590

Since `timestep` is not used in the code in between these calls, we could probably calculate `timestep` on the fly with something like

```python
class SchedulerCommonTest(unittest.TestCase):
    ...
    def test_scheduler_outputs_equivalence(self):
        ...
        for scheduler_class in self.scheduler_classes:
            ...
            if num_inference_steps is not None and hasattr(scheduler, "set_timesteps"):
                    scheduler.set_timesteps(num_inference_steps)
            ...
            timestep = scheduler.timesteps[-1]
```

There might be an issue for schedulers without a `set_timesteps` method, and probably some other cases I've missed, but I think something like this would be much easier to use when writing tests for new schedulers.

**What use case would this enable or better enable? Can you give us a code example?**

See discussion above.

**Comment by dg845** [comment]
One solution might be to have something like

```python
class SchedulerCommonTest(unittest.TestCase):
    ...
    @property
    def default_timestep(self):
        return 0

    @property
    def default_timestep_2(self):
        return 1
    ...
```

and then use `self.default_timestep` and `self.default_timestep_2` in all of the `SchedulerCommonTest` utility methods and tests. Subclasses of `SchedulerCommonTest` that hold tests for individual schedulers can then override these properties if they need to.

**Comment by patrickvonplaten** [comment]
@yiyixuxu can you take a look here?

**Comment by yiyixuxu** [comment]
@patrickvonplaten 
@dg845 is working on it here https://github.com/huggingface/diffusers/pull/5479/files
